scripts/runtests.py

```diff
@@ -203,7 +194,7 @@ def main() -> None:
         print(colored("\n\n--- TEST SUMMARY: One or more tests failed. See above for details. ---\n", "red"))
     else:
         print(colored("\n\n--- TEST SUMMARY: All tests passed! ---\n", "green"))
-    print("Flake8:", _SUCCESS if flake8_result.returncode == 0 else _FAILED)
+    print("pre-commit:", _SUCCESS if pre_commit_result.returncode == 0 else _FAILED)
```

Comment:
Good point. I think it's useful for black and ruff to give nonzero exit codes when they do autofixes, because it's sometimes important to manually review the output of the autofixes before making the PR. What do you think of 6f079ee96c7aa7c39eaf8d56692203085605d5d9?

> idk how this should interact with git pre-commit hooks if `pre-commit` is installed as one, given I don't use 'em.

`¯\_(ツ)_/¯` (I don't use them either 😄)

Reply:
Example output from runtests.py if pre-commit fails, following https://github.com/python/typeshed/commit/6f079ee96c7aa7c39eaf8d56692203085605d5d9:

![image](https://github.com/python/typeshed/assets/66076021/3f1632cb-7546-4b5a-8501-93a03605bc2b)

Reply:
> I think it's useful for black and ruff to give nonzero exit codes when they do autofixes, because it's sometimes important to manually review the output of the autofixes before making the PR. What do you think of [6f079ee](https://github.com/python/typeshed/commit/6f079ee96c7aa7c39eaf8d56692203085605d5d9)?

I think that's valid and something we already could've done by showing a warning instead of passing, for example.
Now with pre-commit, the next best thing is probably failing with a message like you did.


One tiny nitpick is that `Check the output of pre-commit for more details.` might be overly verbose given ` One or more tests failed. See above for details.` is written two lines above. But that won't block my approval

Reply:
Alex finding out the hard way that Terminal is transparent on Macs...

Reply:
WHY CAN'T THERE BE A BUTTON ON THE KEYBOARD FOR PRINT-SCREENING LIKE ON WINDOWS